packages/fx-core/src/plugins/resource/bot/functionsHostedBot/plugin.ts

```diff
@@ -15,6 +15,41 @@ import { TeamsBotImpl } from "../plugin";
 export class FunctionsHostedBotImpl extends TeamsBotImpl {
   public async scaffold(context: PluginContext): Promise<FxResult> {
     this.ctx = context;
-    throw new Error("not implemented");
+
+    await this.config.restoreConfigFromContext(context);
+    this.config.scaffold.hostType = HostTypes.AZURE_FUNCTIONS;
+
+    Logger.info(Messages.ScaffoldingBot);
+
+    const handler = await ProgressBarFactory.newProgressBar(
+      ProgressBarConstants.SCAFFOLD_FUNCTIONS_NOTIFICATION_TITLE,
+      ProgressBarConstants.SCAFFOLD_FUNCTIONS_NOTIFICATION_STEPS_NUM,
+      this.ctx
+    );
+    await handler?.start(ProgressBarConstants.SCAFFOLD_FUNCTIONS_NOTIFICATION_STEP_START);
+
+    // 1. Copy the corresponding template project into target directory.
+    const group_name = TemplateProjectsConstants.GROUP_NAME_BOT;
+    if (!this.config.actRoles || this.config.actRoles.length === 0) {
```

Comment:
if this.config.actRoles.length === 0, then !this.config.actRoles will return false at the first time

Reply:
Fixed